pypeit/edgetrace.py

```diff
@@ -748,14 +748,14 @@ def auto_trace(self, img, bpm=None, det=1, binning=None, save=False, debug=False
         if show_stages:
             self.show(thin=10, include_img=True, idlabel=True)
 
-        # Add user traces
-        if self.par['add_slits'] is not None:
-            self.add_user_traces(trace.parse_user_slits(self.par['add_slits'], self.det))
-
         # Remove user traces
         if self.par['rm_slits'] is not None:
             self.rm_user_traces(trace.parse_user_slits(self.par['rm_slits'], self.det, rm=True))
 
+        # Add user traces
```

Comment:
I can see why one may wish to add after removing, but just give a quick reason here as an inline-doc.  So that we don't revert it! ;)

Reply:
Currently, `sync` isn't run again after slits are removed/added.  However, both the `rm_user_traces` and `add_user_traces` runs `check_synced`, which currently barfs if the traces aren't synced after the traces are added/removed.  I think that behavior makes sense given that you're probably running pypeit fairly interactively if you're requesting traces be added or removed. (But it doesn't allow you to add/rm traces before the object is synced...)  There's a complication, though: When the slits are added, they're added as unlinked traces that are resorted in with the existing traces.  If the input isn't right, this mean you could end with the removed/added traces getting mixed in with existing good slits.  So things are fiddley, and I'm not sure resyncing would get you what you want.

Reply:
To be fair, this update isn't needed for WHT/ISIS - it was a bug that I identified when working on the interactive slits tool. In the various cases I tried, switching the removal/addition of slits didn't cause an error, it was only when slits were added and then removed that I noticed a problem. Probably we can get away without the sync operation. I haven't seen any problems on my end regarding syncing, but I suspect this may need to be looked at for multislit data. @jhennawi - you are correct - the WHT/ISIS detector is fully illuminated.

So, I still propose that the order is switched (makes sense if a user wants to _replace_ a slit (i.e. delete a bad slit and add a good slit). Let's leave out the sync operation until we find it's an issue.

Reply:
Agreed.